samples/bookinfo/networking/bookinfo-gateway.yaml

```diff
@@ -11,7 +11,7 @@ spec:
       name: http
       protocol: HTTP
     hosts:
-    - "*"
+    - "default/*"
```

Comment:
Problem with this is the YAML itself is not namespace bound. So they can apply anywhere, and then this won't work.

Maybe we should have allowed `./*` in the code...

Reply:
Yep, an alternative is to explicitly specify the namespaces for the YAMLs so that they can be *bound* together. The namespace to bind should align w/ the sample READMEs.

In this way, when `kubectl apply` w/ no explicit ns set, the resources will by default be applied to the exact ns specified in the Pod spec. And when `kubectl apply` w/ explicit `-n <ns_name>`, there'll be an error returned to users when they try to apply it elsewhere (message like `the namespace from the provided object "istio-system" does not match the namespace "default". You must pass '--namespace=istio-system' to perform this operation.`), which is guaranteed by kubectl.

Does this approach make sense to you @howardjohn? Thanks!

Reply:
Some of these resources (e.g., bookinfo ones) are intended to be used in whatever namespace the user wants. Even the Istio tasks use different namepaces (e.g., some use default, others use ns bookinfo). Outside blogs and other docs also rely on the namespace flexibility, so this would be unacceptable breaking change IMO.

Reply:
Thanks for the comments!

If it's the case, I think maybe we should have allowed `*` in the code. We should only apply ns scoping to the samples where the YAMLs have already bound w/ specific ns, e.g. `samples/bookinfo/networking/certmanager-gateway.yaml`, `samples/multicluster` etc.

WDYT @frankbu @howardjohn?

Reply:
> We should only apply ns scoping to the samples where the YAMLs have already bound w/ specific ns

That makes sense but I don't think there are very many of those. The `samples/multicluster` east-west gateways are not always in `istio-system`. For example, see the external control plane doc: https://preliminary.istio.io/latest/docs/setup/install/external-controlplane/#setup-east-west-gateways

Reply:
btw, another recommended best practice is to use fully-qualified hosts in VirtualServices (e.g., `reviews.default.svc.cluster.local` insead of just `reviews`), but we haven't done that in the samples either, for the reason reason.

Maybe we need to add more [best practices](https://preliminary.istio.io/latest/docs/ops/best-practices/traffic-management/) to the docs and put some tips in the tasks/examples that point to them.